test/models/retrieve_test.rb

```diff
@@ -1,13 +1,17 @@
 require 'test_helper'
 
 class RetrieveTest < ActiveSupport::TestCase
-  test 'uses correct index for V1 queries' do
-    ClimateControl.modify(V2: nil) do
-      assert_equal ENV.fetch('ELASTICSEARCH_INDEX'), Retrieve.new.default_index
-    end
+  def setup
+    @test_strategy = Flipflop::FeatureSet.current.test!
```

Comment:
Interesting. I wonder if this would have allowed us to avoid the annoying excludev1/v2 split for the past several months  :)

Reply:
I got the idea from the graphql controller tests, so I'm guessing the weirdness persists either way. 🤷